velox/exec/HashTable.cpp

```diff
@@ -1750,18 +1750,63 @@ void HashTable<ignoreNullKeys>::prepareJoinTable(
   }
 }
 
+template <bool ignoreNullKeys>
+inline uint64_t HashTable<ignoreNullKeys>::joinProjectedVarColumnsSize(
+    const std::vector<vector_size_t>& columns,
+    const char* row) const {
+  uint64_t totalBytes{0};
+  for (const auto& column : columns) {
+    if (!rows_->columnTypes()[column]->isFixedWidth()) {
+      totalBytes += rows_->variableSizeAt(row, column);
```

Comment:
I am a little worried about performance implication on this line.  Usually we don't load the row container memory while listing the join results, so the memory is cold and reading from it takes very long time.  Is it possible to do some row size estimation based on total size in row container and adjust the number of listing rows smartly?

Reply:
Thanks @Yuhta , yeah there is going to be some regression, but only on list results part. Quickly after this list results, we are going to do memory copy which is going to be considerably more expensive than list results, making the regression less significant overall. Row size estimation based on total size in row container will not work in this case because we don't know which build side row the probe side is going to match. It could happen that all probe side rows are matching with a few super large build side rows (hence significant skew).

Reply:
We can try to use the average row size of the build size.  Skewness on the build side is hard to solve in this case, but do we run into that in real workload?

Reply:
Yeah, real production workload leads to this improvement.

Reply:
How about we going over the row container to get the maximum row size and use that to adjust max num rows out?  That way we don't need to stride through the memory twice and keep string columns on fast path.

Reply:
Can we also store the RowContainer statistic when we storeRows? And we can also track other statistic like this PR needs https://github.com/facebookincubator/velox/pull/11272